core/src/main/java/org/apache/carbondata/core/util/GeoHashDefault.java

```diff
@@ -0,0 +1,336 @@
+    }
+
+    /* Set the generated column data type as long */
+    properties.put(TARGET_DATA_TYPE, "long");
+
+    // String originLongitude = properties.get(ORIGIN_LONGITUDE);
+    String originLatitude = properties.get(ORIGIN_LATITUDE);
+    if (originLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s property.",
+                      CarbonCommonConstants.INDEX_HANDLER, ORIGIN_LATITUDE));
+    }
+
+    String minLongitude = properties.get(MIN_LONGITUDE);
+    String maxLongitude = properties.get(MAX_LONGITUDE);
+    String minLatitude = properties.get(MIN_LATITUDE);
+    String maxLatitude = properties.get(MAX_LATITUDE);
+    if (minLongitude == null || maxLongitude == null
+            || minLatitude == null || maxLatitude == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. Must specify %s, %s, %s and %s properties.",
+                      CarbonCommonConstants.INDEX_HANDLER, MIN_LONGITUDE, MAX_LONGITUDE,
+                      MIN_LATITUDE, MAX_LATITUDE));
+    }
+
+    String gridSize = properties.get(GRID_SIZE);
+    if (gridSize == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, GRID_SIZE));
+    }
+
+    String conversionRatio = properties.get(CONVERSION_RATIO);
+    if (conversionRatio == null) {
+      throw new MalformedCarbonCommandException(
+              String.format("%s property is invalid. %s property must be specified.",
+                      CarbonCommonConstants.INDEX_HANDLER, CONVERSION_RATIO));
+    }
+
+    /* Fill the values */
+    // this.oriLongitude = Double.valueOf(originLongitude);
+    this.oriLatitude = Double.valueOf(originLatitude);
+    this.userDefineMaxLongitude = Double.valueOf(maxLongitude);
+    this.userDefineMaxLatitude = Double.valueOf(maxLatitude);
+    this.userDefineMinLongitude = Double.valueOf(minLongitude);
+    this.userDefineMinLatitude = Double.valueOf(minLatitude);
+    this.gridSize = Integer.parseInt(gridSize);
+    this.conversionRatio = Integer.parseInt(conversionRatio);
+
+    // 角度转弧度 radians = (Math.PI / 180) * degrees 求得原点纬度角度的余弦数值
+    this.mCos = Math.cos(this.oriLatitude * Math.PI / CONVERT_FACTOR);
+    // 求得 δx=L∗360/(2πR∗cos(lat))
+    this.deltaX = (this.gridSize * 360) / (2 * Math.PI * EARTH_RADIUS * this.mCos);
+    this.deltaXByRatio = this.deltaX * this.conversionRatio;
+    // 求得 δy=L∗360/2πR
+    this.deltaY = (this.gridSize * 360) / (2 * Math.PI * EARTH_RADIUS);
+    this.deltaYByRatio = this.deltaY * this.conversionRatio;
+    // 计算补齐区域并计算栅格i,j表示栅格编号
+    // Xmax = x0+(2^n∗δx) Ymax = y0+(2^n∗δx) 其中N是切的刀数
+    // 其中x0,y0是给定区域的最小x，y坐标， Xmax >= maxLongitude Ymax >= maxLatitude
+    // 计算过程先把maxLongitude, maxLatitude 代入计算出N，如果N不是整数，则取N的下一个整数，代入后求得Xmax，Ymax。
```

Comment:
Please write English comments, not everyone understands Chinese :)

Reply:
Agreed. This code is removed from this PR and it will be raised by @MarvinLitt in a different PR as part algorithm.